go/vt/vtgate/executor.go

```diff
@@ -841,28 +841,56 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 		}, nil
 	case "vitess_tablets":
 		var rows [][]sqltypes.Value
-		stats := e.scatterConn.healthCheck.CacheStatus()
-		for _, s := range stats {
-			for _, ts := range s.TabletsStats {
-				state := "SERVING"
-				if !ts.Serving {
-					state = "NOT_SERVING"
+		if *GatewayImplementation == GatewayImplementationDiscovery {
+			stats := e.scatterConn.GetLegacyHealthCheckCacheStatus()
```

Comment:
scatterConn already had a reference to healthcheck in the legacy implementation. We are not changing that in this iteration because we want to minimize how much we touch the legacy implementation. However, I do see that we can simplify this so that there is less code repetition in executore.

Reply:
actually.. unless we make LegacyTabletsCacheStatusList and TabletsCacheStatusList implement the same interface, I don't see how to avoid the duplicated code.

Reply:
I agree, I leave it up to you.

Reply:
My thinking is that leaving in ugly if-else blocks will force us to clean it up soon :)